nautobot/dcim/models/devices.py

```diff
@@ -998,3 +1013,69 @@ def to_csv(self):
             self.domain,
             self.master.name if self.master else None,
         )
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "dynamic_groups",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "statuses",
+    "webhooks",
+)
+class RedundancyGroup(PrimaryModel, ConfigContextModel, StatusModel):
+    """
+    A RedundancyGroup represents a logical grouping of physical hardware for the purposes of indicating redundancy pairs of high-availability.
+    """
+
+    name = models.CharField(max_length=100, unique=True)
+
+    failover_strategy = models.CharField(
+        max_length=50, blank=True, choices=RedundancyGroupFailoverStrategyChoices, verbose_name="Failover strategy"
+    )
+
+    comments = models.TextField(blank=True)
+
+    secrets_group = models.ForeignKey(
+        to="extras.SecretsGroup",
+        on_delete=models.SET_NULL,
+        default=None,
+        blank=True,
+        null=True,
+    )
```

Comment:
What's the significance/usage of this FK? I don't recall secrets coming up in our design discussions for this model, and I'm not seeing anywhere in the code that's actually *using* this at present. YAGNI?

Reply:
Stubbed out here for the migration (so I didn't have to rebuild) but yes I need to implement the rest of the UIs to show it.

The reason I added it is I thought there might be a shared key for HA between the devices.

Thoughts @lampwins?

Reply:
Yeah since we have the functionally already there to support this let's move forward with it, unless it presents significant effort.